cloud/shared/bin/lib/config_loader.py

```diff
@@ -0,0 +1,77 @@
+import os
+import json
+
+"""
+Config Loader
+  Handles validating and getting data from the configuration/variable files
+  Inputs:
+    - config_filename: a json object with keys being the 
+    variable names and values being the values to set
+    - variable_definition_filename: a json object with keys being the 
+    variable names and values being an metadata object about that 
+    variable (see the variable_definitions.json in the /shared)
+  Call load_files to 
+"""
+
+
+class ConfigLoader:
+    def __init__(self, config_filename, variable_definition_filename):
+        self.config_filename = config_filename
+        self.variable_definition_filename = variable_definition_filename
+
+    def load_files(self):
+        with open(self.config_filename, "r") as config_file:
+            self.configs = json.loads(config_file.read())
+        with open(self.variable_definition_filename, "r") as definitions_file:
+            self.variable_definitions = json.loads(definitions_file.read())
+
+    def __validate_config(self, variable_definitions, configs):
+        is_valid = True
+        validation_errors = []
+
+        for name, definition in variable_definitions.items():
+            is_required = definition.get("required", False)
+            config_value = configs.get(name, None)
+
+            if is_required and config_value is None:
+                is_valid = False
+                validation_errors.append(f"{name} is required, but not provided")
+
+            is_enum = definition.get("type") == "enum"
+
+            if config_value is not None and is_enum:
+                if config_value not in definition.get("values"):
+                    is_valid = False
+                    validation_errors.append(
+                        f"{config_value} not supported enum for {name}"
+                    )
+
+        return is_valid, validation_errors
+
+    def validate_config(self):
+        return self.__validate_config(self.variable_definitions, self.configs)
+
+    def get_cloud_provider(self):
+        return self.configs.get("CIVIFORM_CLOUD_PROVIDER")
+
+    def get_email_sender(self):
```

Comment:
suggestion (blocking): cache the getters
```suggestion
from functools import cache
...
    @cache
    def get_email_sender(self):
```

Reply:
this wasn't in my version of python, i think for the time being the gets should be pretty easy and not need a cache?